subjects/graphql/README.md

```diff
@@ -0,0 +1,185 @@
+
+- **Results table**:
+
+  Both progress and result table will give you the student progression
+
+  | id  |             createdAt |             updatedAt | grade | progressId |                        attrs |
+  | --- | --------------------: | --------------------: | ----: | ---------: | ---------------------------: |
+  | 1   | 2019-07-06T13:52:5... | 2019-07-06T13:52:5... |     0 |         58 |                         `{}` |
+  | 2   | 2019-07-06T13:52:5... | 2019-07-06T13:52:5... |     0 |         58 | `{"errors": {"steps": ...}}` |
+  | 3   | 2019-07-06T13:52:5... | 2019-07-06T13:52:5... |     1 |         58 |                         `{}` |
+
+- **Object table**:
+
+  This table will give you information about all objects (exercises/projects)
+
+  | id  | name |     type | status |                     attrs | childrenAttrs |             createdAt |             updatedAt |
+  | --- | ---: | -------: | -----: | ------------------------: | ------------: | --------------------: | --------------------: |
+  | 1   |    0 | exercise |  draft | `{"language": "dom", ...` |          `{}` | 2019-03-14T12:02:2... | 2019-03-14T12:02:2... |
+  | 2   |    0 |  project | online |  `{"language": "go", ...` |          `{}` | 2019-03-14T12:02:2... | 2019-03-14T12:02:2... |
+  | 3   |    1 | exercise | online |  `{"language": "js", ...` |          `{}` | 2019-03-14T12:02:2... | 2019-03-14T12:02:2... |
+
+Lets take for instance the table `user` and try to query it:
+
+```js
+{
+  query {
+      user {
+          id
+      }
+  }
+}
+```
+
+This simple query will return an array with the `ids` of the users. Imagine if you wanted the `githubLogin`,\
+you could just add this attribute to the query, example:
+
+```js
+{
+  query {
+      user {
+          id
+          githubLogin
+      }
+  }
+}
+```
+
+You can try to `curl` the API endpoint to see the result given by the server:
+
+- `curl "https://[[DOMANIN]]/api/graphql-engine/v1/graphql" --data '{"query":"{user{id githubLogin}}"}'`
+
+Here is another example of a query using the table `user`:
+
+```js
+{
+  query:  {
+    user(where: { id: { _eq: 6 }}) {
+      id
+      githubLogin
+    }
+  }
+}
+```
+
+**Note** that for this query is required the introduction of variables (arguments)\
+ so it will return just one user, the user that as the id equal to `6`.
+
+You can see the result using `curl`:
+
+- `curl "https://[[DOMANIN]]/api/graphql-engine/v1/graphql" --data '{"query":"{user(where:{id:{_eq:6}}){id githubLogin}}"}'`
+
+In graphql the usage of arguments can be specified in the schema of the API endpoint. Here is the _docs_ for the graphql endpoint you are going to use, _https://[[DOMAIN]]/public/subjects/grapqhl_
+
+nesting fetching, using the :
+
+```js
+{
+  result {
+    id
+    user {
+      id
+      githubLogin
+    }
+  }
+}
+
+```
+
+For this example we ask for the results `id` and `user`s that are associated to the `result`, requesting the users `name`s and `id`s.\
```

Comment:
I think instead of talking that much about `graphQL` we should also talk about the UI a bit, maybe ask for them to generate statistic graph using `SVG` and maybe to a bit about hosting ?
where / how should they host it ?

Since xavier was against rolling our own githubpage, maybe we can ask them to publish it on github-page ? or netlify or some other free static hosting service of their choice.